### PR TITLE
core: add deprecation message for disable-device-emulation flag

### PR DESCRIPTION
**Pull request by wardpeet**
**Summary**
Adding small deprecation message for disableDeviceEmulation flag.

For more info chat with @patrickhulce 😄 

**Related Issues/PRs**
https://github.com/GoogleChrome/lighthouse/pull/8289#issuecomment-483693944

### CODE REVIEW

**Comment by patrickhulce** [comment]
It just occurred to me that we can do this after it's been removed too :)

I propose tweaking the text slightly and cherry-picking this back to v4.3.1 if we ever publish a patch release

File: lighthouse-cli/bin.js [hunk 1]
```diff
@@ -94,6 +94,12 @@ async function begin() {
     cliFlags.outputPath = 'stdout';
   }
 
+  // show deprecation message for disableDeviceEmulation
+  if (cliFlags.disableDeviceEmulation) {
+    log.warn('config', 'The "--disable-device-emulation" option is deprecated.' +
```
Comment (patrickhulce) [inline lighthouse-cli/bin.js]: we can make this pull double duty if we say `"--disable-device-emulation" has been removed in v5.`

Comment (wardpeet) [reply lighthouse-cli/bin.js]:
```js
log.warn('config', 'The "--disable-device-emulation" option is deprecated.' +
    ' Please use "--emulated-form-factor=none" instead.');
```

**Comment by wardpeet** [comment]
Feel free to keep this PR around until a patch is going to be made or just close if seems unecessary.

File: lighthouse-cli/bin.js [hunk 2]
```diff
@@ -94,6 +94,12 @@ async function begin() {
     cliFlags.outputPath = 'stdout';
   }
 
+  // show deprecation message for disableDeviceEmulation
```
Comment (brendankenny) [inline lighthouse-cli/bin.js]: maybe add a note we should remove for v6? (we don't have to remove it then, but then we'll know it's alright to at any time after that :)

File: lighthouse-cli/bin.js [hunk 3]
```diff
@@ -94,6 +94,12 @@ async function begin() {
...
+  if (cliFlags.disableDeviceEmulation) {
```
Comment (brendankenny) [inline lighthouse-cli/bin.js]: I think you'll have to `// @ts-ignore disableDeviceEmulation has been removed from cliFlags` this line now that #8289 has removed it

**Comment by wardpeet** [comment]
@brendankenny we could go for something more in your face https://www.npmjs.com/package/boxen

**Comment by brendankenny** [comment]
> @brendankenny we could go for something more in your face https://www.npmjs.com/package/boxen. we could also pause and put some new lines below

sounds terrible, let's not do that :D

**Comment by brendankenny** [comment]
@patrickhulce does this work for you?

**Pull request merged**